model/task/db.go

```diff
@@ -215,36 +217,36 @@ var (
 					},
 					"then": evergreen.TaskTimedOut,
 				},
-				// A task will run if it is activated and does not have any blocking deps
+				// A task will be unscheduled if it is not activated
 				{
 					"case": bson.M{
 						"$and": []bson.M{
+							{"$eq": []interface{}{"$" + ActivatedKey, false}},
 							{"$eq": []string{"$" + StatusKey, evergreen.TaskUndispatched}},
-							{"$eq": []interface{}{"$" + ActivatedKey, true}},
-							{"$eq": []interface{}{isAttainable, true}},
 						},
 					},
-					"then": evergreen.TaskWillRun,
+					"then": evergreen.TaskUnscheduled,
 				},
-				// A task will be unscheduled if it is not activated
+				// A task will be blocked if it has dependencies that are not attainable
 				{
 					"case": bson.M{
 						"$and": []bson.M{
-							{"$eq": []interface{}{"$" + ActivatedKey, false}},
 							{"$eq": []string{"$" + StatusKey, evergreen.TaskUndispatched}},
+							{"$eq": []interface{}{isAttainable, false}},
 						},
 					},
-					"then": evergreen.TaskUnscheduled,
+					"then": evergreen.TaskStatusBlocked,
 				},
-				// A task will be blocked if it has dependencies that are not attainable
+				// A task will run if it is activated and does not have any blocking deps
 				{
 					"case": bson.M{
 						"$and": []bson.M{
 							{"$eq": []string{"$" + StatusKey, evergreen.TaskUndispatched}},
-							{"$eq": []interface{}{isAttainable, false}},
+							{"$eq": []interface{}{"$" + ActivatedKey, true}},
+							{"$eq": []interface{}{isAttainable, true}},
```

Comment:
This check seems superfluous since we've already established the task isn't blocked in the previous case.